### PR TITLE
Add code for action

### PR DESCRIPTION
**Pull request by chelnak**
This commit adds the initial implementation for the community labeler action.

It will most probably change based on reviews and suggestions from the team.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+/* When an issue or pull request is opened we want to look at the associated user and
+see if they are a member of the X organisations and Y teams. If they are not we should
+label the issue with the label `community`. */
+async function run(): Promise<void> {
+  try {
+    // Retrieve our inputs
+    const labelName = core.getInput('label_name', {required: true})
+    const labelColor = core.getInput('label_color', {required: true})
+    const loginsToIgnore = core.getInput('logins_to_ignore', {required: false})
+    const orgs = core.getInput('orgs', {required: true}).split(',')
```
Comment (petergmurphy) [inline src/main.ts:13]: Should this be `org_membership` instead of `orgs`?

**Comment by chelnak** [comment]
Two thoughts on this..

1. Should we be posting some sort of comment on issues and prs that we label?

```
Hello 👋 

We've labeled this issue/pr as a community contribution. ( Maybe something here about why we are labeling?)

If you think that this label has been applied in error, let us know (link to action repo issues) !
```

2. It needs some sort of guard clause to stop it from re-labeling issues

File: src/main.ts (hunk 1 above)
Comment (chelnak) [reply src/main.ts]: I've resolved these two comments & added a test for the scenario.

Thanks!

**Comment by petergmurphy** [comment]
> Two thoughts on this..
> 
> 1. Should we be posting some sort of comment on issues and prs that we label?
> 
> ```
> Hello 👋 
> 
> We've labeled this issue/pr as a community contribution. ( Maybe something here about why we are labeling?)
> 
> If you think that this label has been applied in error, let us know (link to action repo issues) !
> ```
> 
> 2. It needs some sort of guard clause to stop it from re-labeling issues

Anyone watching the thread will be notified when the `community` label is attached to the issue right? Although it might be good as it re-engages the creator of the issue, which may lead to the creator closing the issue if it's outdated.

**Review by petergmurphy — APPROVED** [review]

**Pull request merged**